msgp/extension.go

```diff
@@ -76,7 +76,7 @@ type Extension interface {
 	// type of the extension. (Types <0 are
 	// officially reserved by the MessagePack
 	// specifications.)
-	ExtensionType() int8
+	ExtensionType() uint8
```

Comment:
1) You cannot change an exported interface.
2) Extension types are defined as signed in spec. Why this change at all?